client/src/app/template-questionnaire/template-generator.service.ts

```diff
@@ -35,28 +36,25 @@ export class TemplateGeneratorService {
   }
 
   generateTemplate(templateSpec: TemplateSpecification): Observable<Blob> {
-    return Observable
-      .from(this._requestToGenerate(templateSpec))
-      .flatMap(data =>
-        Observable.interval(this.POLLING_INTERVAL)
-          .flatMap(() => this._requestToDownload(data._links.download.href))
-          .filter(template => template.complete)
-          .take(1)
-          .map(template => template.data)
-          .timeoutWith(this.TIMEOUT,
-            throwError(new Error('Retrieval of template spreadsheet has timed out.'))
-          ));
+    return timer(0, this.POLLING_INTERVAL).pipe(
```

Comment:
@MightyAx Quick question - it looks like the behaviour here is different to before, it attempts to generate a new spreadsheet every `POLLING_INTERVAL`, whereas the old code attempted to download the spreadsheet every `POLLING_INTERVAL`. Is this change intentional?

Reply:
Yes good catch, I hadn't realised that the way that `_requestToGenerate` and `_requestToDownload` were linked once call to `_requestToGenerate` will require polling to `_requestToDownload` until the file is generated.

This was on my list to test though 😅